File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
import numbers
import math


def is_armstrong(number):
    check_number(number)
    sum = 0
    for nmbr in get_digits(number):
        sum += math.pow(nmbr, get_number_of_digits(number))
    if number == sum: return True
    return False


def get_digits(number):
    check_number(number)
    digits = []
    if number == 0:
        return digits
    while number > 0:
        digits.append(number%10)
        number = int(number / 10)
    digits.reverse()
    return digits


def get_number_of_digits(number):
    check_number(number)
    if number == 0: return 1
    return int(math.log(number, 10)) + 1


def check_number(number):
    if not isinstance(number, numbers.Number):
        raise ValueError('limit has to be a number')
    elif number < 0:
        raise ValueError('number must be greater than -1')

File: solutions/typescript/armstrong-numbers/1/armstrong-numbers.ts
export default class ArmstrongNumbers {
    static isArmstrongNumber(input: number): boolean {
        // console.log("input: ", input);
        const numberOfDigits = ArmstrongNumbers.getNumberOfDigits(input);
        // console.log("numberOfDigits: ", numberOfDigits);
        const digitsArray = ArmstrongNumbers.getDigits(input);
        // console.log("digitsArray: ", digitsArray);
        const reducer = (accumulator: number, currentValue: number) => accumulator + (Math.pow(currentValue, numberOfDigits));
        const armstrongNumberCandidate = digitsArray.reduce(reducer, 0);
        // console.log("armstrongNumberCandidate: ", armstrongNumberCandidate);
        return (input > -1 ? input === armstrongNumberCandidate : false);
    }

    private static getNumberOfDigits(input: number): number {
        return Math.floor(Math.log10(input)+1);
    }

    private static getDigits(input: number): number[] {
        let nmbr = input;
        let digits:number[] = []; 
        if(nmbr>0){
            while (nmbr > 0) {
                digits.push(Math.floor(nmbr % 10));
                nmbr = Math.floor(nmbr / 10);
            }
            digits = digits.reverse();
        } else if (nmbr === 0) {
            digits.push(0);
        }
        return digits;
    }
}
  